selfdrive/hardware/tici/agnos.py

```diff
@@ -120,17 +172,23 @@ def flash_partition(cloudlog, spinner, target_slot, partition):
 
     # Write hash after successfull flash
     os.sync()
-    out.write(partition['hash_raw'].lower().encode())
+    if not full_check:
+      out.write(partition['hash_raw'].lower().encode())
 
 
-def flash_agnos_update(manifest_path, cloudlog, spinner=None):
+def swap(manifest_path, target_slot_number):
   update = json.load(open(manifest_path))
+  for partition in update:
+    if not partition.get('full_check', False):
+      clear_partition_hash(target_slot_number, partition)
+
+  os.system(f"abctl --set_active {target_slot_number}")
```

Comment:
return code not checked. should we make a nice abctl wrapper or reimplement in python?
```suggestion
  subprocess.check_output(f"abctl --set_active {target_slot_number}", shell=True)
```

Reply:
We should reimplement it, it's very slow right now.